### PR TITLE
[skip ci]convert PR GATE into pipeline

### PR DESCRIPTION
**Pull request by PengTian0**
Put Jenkins Jobs for "PR Gate" under source code control.
Depends on https://github.com/RackHD/on-build-config/pull/104,  https://github.com/RackHD/on-build-config/pull/91,  https://github.com/RackHD/on-build-config/pull/90

Test:
A test running of the pipeline which is triggered by a PR of on-http which depends on a PR of on-tasks):
image

A test running of the pipeline  which is triggered by a PR of on-core:
image

### CODE REVIEW

**Comment by PengTian0** [comment]
@panpan0000 @changev @anhou @yyscamper @keedya

File: jobs/unit_test/unit_test.groovy [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                        node(node_name){
+                            deleteDir()
+                            dir("build-config"){
+                                checkout scm
+                            }
+                            unstash 'unittest_manifest'
+                            timeout(15){
+                                try{
+                                    sh "./build-config/jobs/unit_test/unit_test.sh ${repo_name}"
+                                } catch(error){
```
Comment (panpan0000) [inline jobs/unit_test/unit_test.groovy:59]: nice

File: jobs/unit_test/unit_test.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    --builddir ${WORKSPACE}/build-deps \
+    --jobs 8 \
+    --force \
+    checkout \
+    packagerefs
+    if [ -d "build-deps/on-build-config" ]; then
+        rm -rf build-config
+        cp -r build-deps/on-build-config build-config
+    fi
+    mongo pxe --eval "db.dropDatabase()"
```
Comment (panpan0000) [inline jobs/unit_test/unit_test.sh:17]: AHa!

**Review by panpan0000 — APPROVED** [review]
when you paste the enough sandbox test result, then it's a 👍

**Review by anhou — APPROVED** [review]

**Pull request merged**